### PR TITLE
Clean up pool asset

### PR DESCRIPTION
**Pull request by hieuvubk**
Closes: #1959

## What is the purpose of the change

Removed dead code 
Type declaration moved to the top of the file
Linked methods are unexported
Remove TODO

### CODE REVIEW

**Review by alexanderbez — APPROVED** [review]

File: x/gamm/pool-models/balancer/pool_asset.go [hunk 1]
```diff
@@ -12,40 +12,27 @@ import (
...
+// validates a pool asset, to check if it has a valid weight.
+func (asset PoolAsset) validateWeight() error {
 	if asset.Weight.LTE(sdk.ZeroInt()) {
 		return fmt.Errorf("a token's weight in the pool must be greater than 0")
 	}
 
-	// TODO: Choose a value that is too large for weights
-	// if asset.Weight >= (1 << 32) {
-	// 	return fmt.Errorf("a token's weight in the pool must be less than 2^32")
-	// }
```
[the remark this answers is not in the document]
Comment (hieuvubk) [reply x/gamm/pool-models/balancer/pool_asset.go:24]: I'll create an issue about it. Tried to add a check but some tests were wrong in PokeTokenWeights

[the remark this answers is not in the document]
Comment (p0mvn) [reply x/gamm/pool-models/balancer/pool_asset.go:24]: Just FYI - there is already an issue for this: https://github.com/osmosis-labs/osmosis/issues/1958.

@hieuvubk could you please add a comment here:
```
// TODO: add validation for asset weight overflow:
// https://github.com/osmosis-labs/osmosis/issues/1958
```

**Review by p0mvn — APPROVED** [review]
LGTM - I would only like to ask to keep the comment about overflow validation

**Comment by hieuvubk** [comment]
> LGTM - I would like to only ask to keep the comment about overflow validation

Yes I did. Thanks for the suggestion

**Review by alexanderbez — APPROVED** [review]

**Pull request merged**